wcfsetup/install/files/lib/data/I18nDatabaseObjectList.class.php

```diff
@@ -41,6 +41,9 @@ public function __construct($languageID = null)
         if ($languageID === null) {
             $languageID = WCF::getUser()->languageID;
         }
```

Comment:
I wonder if this fall back should be removed. Either one wants an explicit language selection or one wants the effective current language. I can't imagine a case where the users configured language might be preferred over the language that is currently being in use.

Reply:
The problem was that the user has no languageID set which resulted in invalid SQL. Using this fallback it was fixed.
And I don't want to use a language explicitly every time. Why should I do that by default? By default I want to use the user's or session's language.

Reply:
> The problem was that the user has no languageID set which resulted in invalid SQL.

Yes, I understand that.

> Using this fallback it was fixed.

I specifically commented on lines 41 to 43 (i.e. the `WCF::getUser()->languageID`) part. My understanding is that this value is never the correct value. Either I want an explicit language or the language that's currently in use by the language subsystem (i.e. the lines you added).